core/Tracker/GoalManager.php

```diff
@@ -321,7 +321,7 @@ public function recordGoals(VisitProperties $visitProperties, Request $request)
      */
     protected function getRevenue($revenue)
     {
-        if (round($revenue) != $revenue) {
+        if (round($revenue) !== $revenue) {
```

Comment:
I guess this is another check if we have fraction digits. To also handle integers correctly I think it would need to be
```suggestion
        if (round($revenue) !== (float) $revenue) {
```